src/components/common/messaging/MessageBox.tsx

```diff
@@ -185,12 +188,20 @@ export default observer(({ channel }: Props) => {
             }
         }
 
+        const textBox = document.getElementById("message");
+        textBox?.addEventListener("compositionstart", () => {
+            setComposed(false);
+        });
+        textBox?.addEventListener("compositionend", () => {
+            setComposed(true);
+        });
```

Comment:
Can this not just use React event bindings?

Reply:
I tried React's onCompositionStart and onCompositionEnd, but they didn't work in my environment. (There is still a bug in this event, which may be the reason.)
https://github.com/facebook/react/issues/3926
So I am using addEventListener. If I could replace them with onCompositionStart/End, I think that would be better.

Reply:
Any progress on this?